api/v1beta1/hyperconverged_types.go

```diff
@@ -180,6 +180,10 @@ type HyperConvergedSpec struct {
 	// +kubebuilder:validation:Enum=None;LiveMigrate;External
 	// +optional
 	EvictionStrategy *v1.EvictionStrategy `json:"evictionStrategy,omitempty"`
+
+	// VMStateStorageClass is the name of the storage class to use for the PVCs created to preserve VM state, like TPM.
+	// The storage class must support RWX in filesystem mode.
+	VMStateStorageClass string `json:"vmStateStorageClass,omitempty"`
```

Comment:
Can you please use a `*string` and `+optional` for consistency with what we're doing with other optional values here.

Can you please also document it under [cluster-configuration.md](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/docs/cluster-configuration.md) ?